sky/resources.py

```diff
@@ -1290,6 +1291,22 @@ def is_empty(self) -> bool:
     def copy(self, **override) -> 'Resources':
         """Returns a copy of the given Resources."""
         use_spot = self.use_spot if self._use_spot_specified else None
+
+        current_override_configs = self._cluster_config_overrides
+        if self._cluster_config_overrides is None:
+            current_override_configs = {}
+        new_override_configs = override.pop('_cluster_config_overrides', {})
+        overlaid_configs = skypilot_config.overlay_skypilot_config(
+            original_config=config_utils.Config(current_override_configs),
+            override_configs=new_override_configs,
+        )
+        override_configs = config_utils.Config()
+        for key in constants.OVERRIDEABLE_CONFIG_KEYS_IN_TASK:
+            elem = overlaid_configs.get_nested(key, None)
+            if elem:
+                override_configs.set_nested(key, elem)
+
+        override_configs = dict(override_configs) if override_configs else None
```

Comment:
Why do we need the conditional?

Reply:
ah I remember now, it was because a bunch of unittests were failing without this line (see CI results for https://github.com/skypilot-org/skypilot/pull/5197/commits/b06d115b64f40d2f953238170970b234592f34e2)

Reply:
I mean doesn't this work?
```suggestion
        override_configs = dict(override_configs)
```

Reply:
yeah even that fails some tests https://github.com/skypilot-org/skypilot/pull/5197/commits/412ec6d958700c702bf9dd51c1c4aa41cdc157e4